src/components/Room/UsernameForm.tsx

```diff
@@ -25,17 +28,15 @@ function UsernameForm({ handleClose }: UsernameFormProps) {
   // If we have a username in redux slice, start it with that. else, start it with empty string.
   const [inputtedUsername, setInputtedUsername] = useState<string>(username ? username : '');
   const [usernameSubmissionDone, setUsernameSubmissionDone] = useState<boolean>(false);
-
-  const submitHandler = async (event: React.MouseEvent<HTMLButtonElement>) => {
-    event.preventDefault();
-
-    // TODO: Validation - may want to use formik/yup?
-    // Make sure the username they gave isn't empty / made of only white spaces
-    const trimmedInputtedUsername: string = inputtedUsername.trim();
-    if (trimmedInputtedUsername === '') {
-      console.log('cannot have empty username');
-      // stop now! and display error. user needs to resubmit. - Or use formik/yup
-    } else {
+  const [creationFailure, setCreationFailure] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const formik = useFormik({
+    initialValues: {
+      username: '',
```

Comment:
do we want the initialValue to start with the username in redux, if there is one?
This is how, in a previous iteration i allowed for the previous username to be autofilled.
I think this is a design decision we should make - i liked having the name be autofilled because it felt like 'oh this game remembers me!'